# HG changeset patch
# User Jordon Moss <mossj32@gmail.com>
# Date 1744976341 10800
#      Fri Apr 18 08:39:01 2025 -0300
# Node ID 31a8ee9802826d63f67220f06a08e59dbc060143
# Parent  020695547d33a6335060fa28443bc1058614ee8c
# EXP-Topic strikers_stuff
Add APROP_GibHealth.

diff -r 020695547d33 -r 31a8ee980282 docs/zandronum-history.txt
--- a/docs/zandronum-history.txt	Fri Apr 18 07:15:51 2025 -0300
+++ b/docs/zandronum-history.txt	Fri Apr 18 08:39:01 2025 -0300
@@ -26,6 +26,7 @@
 +	- Backport A_FaceTarget flags from ZDoom revision 53fd57d6, and add a new flag, FAF_PITCHFROMORIGIN for A_FaceTarget. [StrikerTheHedgefox]
 +	- Backport APROP_DamageType. [StrikerTheHedgefox]
 +	- Hitting ice corpses now trigger damage events. [StrikerTheHedgefox]
+*	- Added APROP_GibHealth, which lets you retrieve the gib health of an actor. [StrikerTheHedgefox]
 +	- Added APROP_SpawnFlags, which gets/sets the map spawn flags of an actor. Useful for passing these down to other actors. [StrikerTheHedgefox]
 +	- Added APROP_Special, and APROP_Args0-4 to Get/Set/CheckActorProperty. [StrikerTheHedgefox]
 +	- Added A_JumpTo codepointer, which jumps to a state instantly & with prediction. Useful for jumping to an ancestor actor's state without the issues A_Jump with a 256 chance has. [StrikerTheHedgefox]
diff -r 020695547d33 -r 31a8ee980282 src/p_acs.cpp
--- a/src/p_acs.cpp	Fri Apr 18 07:15:51 2025 -0300
+++ b/src/p_acs.cpp	Fri Apr 18 08:39:01 2025 -0300
@@ -4920,6 +4920,9 @@
 		actor->HandleSpawnFlags();
 		break;
 
+	case APROP_GibHealth: // [JM]
+		break; // Can't set this.
+
 	default:
 		// do nothing.
 		break;
@@ -5027,6 +5030,7 @@
 	case APROP_Args3:		return actor->args[3]; // [JM]
 	case APROP_Args4:		return actor->args[4]; // [JM]
 	case APROP_SpawnFlags:	return actor->SpawnFlags; // [JM]
+	case APROP_GibHealth:	return actor->GibHealth();
 
 	default:				return 0;
 	}
@@ -5080,6 +5084,7 @@
 		case APROP_Args2: // [JM]
 		case APROP_Args3: // [JM]
 		case APROP_Args4: // [JM]
+		case APROP_GibHealth: // [JM]
 			return (GetActorProperty(tid, property) == value);
 
 		case APROP_SpawnFlags: // [JM]
diff -r 020695547d33 -r 31a8ee980282 src/p_acs.h
--- a/src/p_acs.h	Fri Apr 18 07:15:51 2025 -0300
+++ b/src/p_acs.h	Fri Apr 18 08:39:01 2025 -0300
@@ -431,6 +431,7 @@
 	APROP_Args3			= 204, // [JM]
 	APROP_Args4			= 205, // [JM]
 	APROP_SpawnFlags	= 206, // [JM]
+	APROP_GibHealth		= 207, // [JM]
 };	
 
 // [Dusk] Enumeration for GetTeamProperty
